app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,13 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rules_and_groups = rules + RuleGroup.where(id: rules.includes(:rule_group).flat_map(&:rule_group_ancestor_ids))
+
+    conflicts = RuleGroupRelationship.where(left: rules_and_groups, relationship: 'conflicts').map(&:right).uniq
+    requires = RuleGroupRelationship.where(left: rules_and_groups, relationship: 'requires').map(&:right).uniq
```

Comment:
This can be definitely simplified, you can do `.select(:right).distinct.map(&:right)` which should be faster as the `uniq` is done in SQL. But I'd move it partially to a scope :thinking: except the map part maybe?

Reply:
Yeah, that should work! I think leaving the map part out of the scope is best too because it'll keep it more general so we can use it again in the future if we need.